python/tskit/combinatorics.py

```diff
@@ -82,6 +106,135 @@ def random_binary_tree(leaf_labels, rng):
             root = root.parent
         return root
 
+    @classmethod
+    def balanced_tree(cls, leaf_labels, arity):
+        """
+        Returns a balanced tree of the specified arity. At each node the
+        leaf labels are split equally among the arity children using the
+        equal_chunks method.
+        """
+        assert len(leaf_labels) > 0
+        if len(leaf_labels) == 1:
+            root = cls(label=leaf_labels[0])
+        else:
+            children = [
+                cls.balanced_tree(chunk, arity)
+                for chunk in equal_chunks(leaf_labels, arity)
+            ]
+            root = cls(children=children)
+            for child in children:
+                child.parent = root
+        return root
+
+
+def generate_star(num_leaves, *, span, branch_length, record_provenance):
+    """
+    Generate a star tree for the specified number of leaves.
+
+    See the documentation for Tree.generate_balanced for more details.
+    """
+    if num_leaves < 2:
+        raise ValueError("The number of leaves must be 2 or greater")
+    tables = tskit.TableCollection(sequence_length=span)
+    tables.nodes.set_columns(
+        flags=np.full(num_leaves, tskit.NODE_IS_SAMPLE, dtype=np.uint32),
+        time=np.zeros(num_leaves),
+    )
+    root = tables.nodes.add_row(time=branch_length)
+    tables.edges.set_columns(
+        left=np.full(num_leaves, 0),
+        right=np.full(num_leaves, span),
+        parent=np.full(num_leaves, root, dtype=np.int32),
+        child=np.arange(num_leaves, dtype=np.int32),
+    )
+    if record_provenance:
+        # TODO replace with a version of https://github.com/tskit-dev/tskit/pull/243
+        # TODO also make sure we convert all the arguments so that they are
+        # definitely JSON encodable.
+        parameters = {"command": "generate_star", "TODO": "add parameters"}
+        tables.provenances.add_row(
+            record=json.dumps(tskit.provenance.get_provenance_dict(parameters))
+        )
+    return tables.tree_sequence().first()
+
+
+def generate_comb(num_leaves, *, span, branch_length, record_provenance):
+    """
+    Generate a comb tree for the specified number of leaves.
```

Comment:
Also called a "pectinate" tree - probably worth including in the docs for search purposes